WordPress/gutenberg
# [RNMobile] - Layout Template Modal & Button block - Orientation fixes

`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2086

## Description
It was [reported](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2064) that the Layout template picker was crashing for the `14.5` version when the Modal was opened and then changing the orientation of the device (**only for Android**).

The orientation crash was also happening for the Inserter modal but these issues were fixed [in this PR](https://github.com/WordPress/gutenberg/pull/20860) to be added in `14.6`.

Although the previously mentioned PR fixed the modals crash it also introduced some side effects for the `Layout template modal` and the `Button block`.

This PR fixes those issues related to orientation changes on Android.

## How has this been tested?

- Open the WordPress Android app with metro running

- **Button block test**
  - Start a new post
  - Add a button block
  - Change the device's orientation
  - Switch back to the previous orientation
  - Expect no link settings to show up

- **Layout preview modal test**
  - Start a new page
  - Select a template
  - Expect to see the modal with the preview (do not close it)
  - Change the device's orientation
  - Expect the modal to disappear and no random modals to show up
  - Switch back to the previous orientation
  - Expect the editor to not crash and the template's options to be tappable

## Screenshots

| Button Block Before | Button Block After |
|     :---:      |     :---:      |
| <img src="https://user-images.githubusercontent.com/4885740/78036002-15f00300-736a-11ea-8c68-40e3812752ca.gif" width=250 /> | <img src="https://user-images.githubusercontent.com/4885740/78036037-230cf200-736a-11ea-97f2-1f0ba6aa3032.gif" width=250 />  |

| Layout Preview Before | Layout Preview After |
|     :---:      |     :---:      |
| <img src="https://user-images.githubusercontent.com/4885740/78036107-37e98580-736a-11ea-9cf6-fa026533d3f6.gif" width=250 /> | <img src="https://user-images.githubusercontent.com/4885740/78036139-3fa92a00-736a-11ea-932c-a97cc7d9865a.gif" width=250 />  |

## Types of changes
Bug fix

## Checklist:
